animal-options.lua

```diff
@@ -0,0 +1,140 @@
+--@ module = true
+
+local overlay = require('plugins.overlay')
+local widgets = require('gui.widgets')
+local gui = require('gui')
+
+-- Make sure an animal unit of your civ is selected as well as not a pet
+local function check_valid_unit()
+    local unit = dfhack.gui.getSelectedUnit(true)
+    if unit ~= nil
+        and dfhack.units.isFortControlled(unit)
+        and dfhack.units.isAlive(unit)
+        and not dfhack.units.isPet(unit)
+        and dfhack.units.isAnimal(unit) then
+
+            return true
+    else
+        return false
+    end
+end
+
+-- The above function already handles checking if valid unit
+-- so just set slaughter flag
+local function set_slaughter_flag()
+    local unit = dfhack.gui.getSelectedUnit(true)
+
+    if unit.flags2.slaughter == 1 then
+        unit.flags2.slaughter = 0
+    else
+        unit.flags2.slaughter = 1
+    end
+end
+
+-- set units marked for gelding flag
+local function set_gel_flag()
+    local unit = dfhack.gui.getSelectedUnit(true)
+
+    -- If the unit is not male or already gelded do nothing
+    if unit.sex ~= 1 or unit.flags3.gelded == 1 then
+        return
+    end
+
+    if unit.flags3.marked_for_gelding == 1 then
+        unit.flags3.marked_for_gelding = 0
+    else
+        unit.flags3.marked_for_gelding = 1
+    end
+end
+
+-- set available for adoption flag
+local function set_adoption_flag()
+    local unit = dfhack.gui.getSelectedUnit(true)
+
+    if unit.flags3.available_for_adoption == 1 then
+        unit.flags3.available_for_adoption = 0
+    else
+        unit.flags3.available_for_adoption = 1
+    end
+end
+
+-- Check current flag status of animal to dynamically set on/off for each unit?
+local function initial_butcher()
+    local unit = dfhack.gui.getSelectedUnit(true)
+
+    if unit.flags2.slaughter == 1 then
+        return true
+    else
+        return false
+    end
+end
+
+local function initial_geld()
+    local unit = dfhack.gui.getSelectedUnit(true)
+
+    if unit.flags3.marked_for_gelding == 1 then
+        return true
+    else
+        return false
+    end
+end
+
+local function initial_adopt()
+    local unit = dfhack.gui.getSelectedUnit(true)
+
+    if unit.flags3.available_for_adoption == 1 then
+        return true
+    else
+        return false
+    end
+end
+
+creature_screen=defclass(creature_screen, overlay.OverlayWidget)
+creature_screen.ATTRS {
+    desc = "Add options to tamed animals view sheet",
```

Comment:
How about also adding an option for assigning a trainer? The logic for that is written (for the `logistics` plugin), but I might have to make it available to lua

Reply:
Interesting after the base capability is flushed out and gets merged in, I'll make it the next thing I work on. Would be nice to just have all the options there since already have 3 of them.